pyomo/solvers/plugins/solvers/MIPCL.py

```diff
@@ -0,0 +1,119 @@
+#  ___________________________________________________________________________
+#
+#  Pyomo: Python Optimization Modeling Objects
+#  Copyright 2017 National Technology and Engineering Solutions of Sandia, LLC
+#  Under the terms of Contract DE-NA0003525 with National Technology and
+#  Engineering Solutions of Sandia, LLC, the U.S. Government retains certain
+#  rights in this software.
+#  This software is distributed under the 3-clause BSD License.
+#  ___________________________________________________________________________
+
+import logging
+
+import pyutilib.services as services
+
+from pyutilib.misc import Bunch, Options
+from pyutilib.services.TempfileManager import create_tempfile
+
+from pyomo.util.plugin import alias
+from pyomo.opt import SolverResults
+from pyomo.opt.base import OptSolver, SolverFactory, ProblemFormat, ResultsFormat
+from pyomo.opt.solver import SystemCallSolver
+from pyomo.opt.base.solvers import _extract_version
+
+from six import iteritems, string_types
+
+logger = logging.getLogger('pyomo.solvers')
+
+_mipcl_version = None
+
+class MIPCL(OptSolver):
+    """The MIPCL LP/MIP solver"""
+
+    alias('mipcl', doc='The MIPCL LP/MIP solver')
+
+    def __new__(cls, *args, **kwds):
+        try:
+            mode = kwds['solver_io']
+            if mode is None:
+                mode = 'mps'
+            del kwds['solver_io']
+        except KeyError:
+            mode = 'mps'
+        if mode == 'mps':
+            opt = SolverFactory('_mipcl_shell', **kwds)
+            opt.set_problem_format(ProblemFormat.mps)
+            return opt
+        if mode == 'os':
+            opt = SolverFactory('_ossolver', **kwds)
+        else:
+            logger.error('Unknown IO type: %s' % mode)
+            return
+        opt.set_options('solver=mps_mipcl')
+        return opt
+
+class MIPCLSHELL(SystemCallSolver):
+    """Shell interface to the MIPCL LP/MIP solver"""
+
+    alias('_mipcl_shell', doc='Shell interface to the MIPCL LP/MIP solver')
+
+    def __init__(self, **kwds):
+        kwds['type'] = 'mipcl'
+        SystemCallSolver.__init__(self, **kwds)
+
+        self._valid_problem_formats = [ProblemFormat.mps]
+        self._valid_result_formats = {ProblemFormat.mps: ResultsFormat.sol}
```

Comment:
`ResultsFormat.soln` is the correct choice here. The `sol` format is an actual standard format used by the AMPL Solver Library, whereas the `soln` format signifies that this plugin uses a solver specific solution parser. It will cause the `process_soln_file` method to be called so that it can add additional solution info to the `SolverResults` object returned by the `process_logfile` method. Although, one _could_ just do all of the processing in either of those methods.